gramex/apps/nlg/body.html

```diff
@@ -0,0 +1,344 @@
+                        <% } %>
+                      </script>
+                    </tbody>
+                  </table>
+                  <div id="srdd"></div>
+                  <button id="check-template" title="Run template"
+                  type="button" class="btn btn-primary btn-sm">
+                      Run Template <i class="fa fa-play"></i>
+                  </button>
+                  <button id="save-template" title="Save Template"
+                  type="button" class="btn btn-primary btn-sm" disabled>
+                      Save Template <i class="fa fa-save"></i>
+                  </button>
+                </div>
+            </div>
+        </div>
+    </div>
+</div>
+
+<!-- traceback modal-->
+<div id="tb-modal" class="modal fade bd-example-modal-lg" tabindex="-1"
+role="dialog" aria-labelledby="myLargeModalLabel" aria-hidden="true">
+  <div class="modal-dialog modal-lg">
+    <div class="modal-content">
+      <div class="modal-header">
+        <h5 class="modal-title">Traceback</h5>
+        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+          <span aria-hidden="true">&times;</span>
+        </button>
+      </div>
+      <div class="modal-body" id="traceback">
+      </div>
+    </div>
+  </div>
+</div>
+<!-- Include JS dependencies  -->
+<script src="{{ nlg_base }}/ui/jquery/dist/jquery.min.js"></script>
+<script src="{{ nlg_base }}/ui/bootstrap/dist/js/bootstrap.bundle.min.js"></script>
+<script src="{{ nlg_base }}/ui/lodash/lodash.min.js"></script>
+<script src="{{ nlg_base }}/ui/g1/dist/g1.min.js"></script>
+<script src="{{ nlg_base }}/ui/bootstrap-select/dist/js/bootstrap-select.min.js"></script>
+<script src="{{ nlg_base }}/nlg.js"></script>
+<script>
+  $.fn.selectpicker.Constructor.BootstrapVersion = '4'
+  var nlg_base = "{{ nlg_base }}"
+  var df = null;
+  var args = null;
+  var templates = [];
+  var currentEventHandlers = {};
+  var currentEditIndex = null;
+  var currentSpanId = null;
+  var grammarOptions = []
+  var narrative_name = "";
+  var dataset_name = "";
+
+  $(document).ready(setInitialConfig)
+
+  $('.formhandler').on('load', renderPreview)
+  $('.formhandler').formhandler()
+
+  $('#save-config-btn').on('click', saveConfig)
+  $('#inspect').on('click', addToNarrative)
+  $('#check-template').on('click', checkTemplate)
+  $("#save-template").on("click", saveTemplate)
+  $("#edit-template").on('focus', (e) => {$('#save-template').attr('disabled', false)})
+  $('#condition-btn').on('click', addCondition)
+  $('#share-narrative-btn').on('click', shareNarrative)
+  $("#copy-editor-url").on("click", (e) => {copyToClipboard('share-editor-url')})
+  $("#copy-narrative-url").on("click", (e) => {copyToClipboard('share-narrative-url')})
+  $('#tmpl-name-btn').on('click', addName)
+  $('#fh-arg-setter').on('change', changeFHSetter)
+  $("#textbox").attr('cols', 119)
+  $("#edit-template").attr('cols', 69)
+  $("#edit-template").attr('rows', 10)
```

Comment:
You can chain last 2 lines.

For events, move them to live event patterns, like

```
$(body)
.on('click', '#save-config-btn', ....)
.on('click', '#inspect', ....)
....
```